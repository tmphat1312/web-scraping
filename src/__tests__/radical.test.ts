import {
  scrapeRadical,
  scrapeRadicalFoundInKanji,
} from '../populating/scripts/radical';

describe('scrape radicals', () => {
  it('should be the same radical', async () => {
    let radical = await scrapeRadical('ground');
    expect(radical).toMatchSnapshot();
  });

  it('should be the same radical found in kanji', async () => {
    let foundInKanji = await scrapeRadicalFoundInKanji('ground');
    expect(foundInKanji).toMatchSnapshot();
  });
});
