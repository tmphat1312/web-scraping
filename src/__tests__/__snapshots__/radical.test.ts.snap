// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`scrape radicals > should be the same radical 1`] = `
{
  "alternative_meanings": undefined,
  "character": "一",
  "level": 1,
  "mnemonic": [
    "This radical consists of a single, horizontal stroke. What's the biggest, single, horizontal stroke? That's the <mark title="Radical" class="radical-highlight">ground</mark>. Look at the ground, look at this radical, now look at the ground again. Kind of the same, right?",
  ],
  "primary_meaning": "Ground",
}
`;

exports[`scrape radicals > should be the same radical found in kanji 1`] = `
{
  "found_in_kanji": [
    "上",
    "下",
    "一",
    "三",
    "正",
    "才",
    "写",
    "号",
    "来",
    "同",
    "両",
    "前",
    "室",
    "屋",
    "具",
    "使",
    "求",
    "便",
    "低",
    "輪",
    "昼",
    "係",
    "真",
    "底",
    "兵",
    "善",
    "論",
    "妻",
    "岡",
    "挙",
    "域",
    "更",
    "師",
    "武",
    "与",
    "再",
    "極",
    "惑",
    "浜",
    "益",
    "並",
    "監",
    "覧",
    "帯",
    "興",
    "宣",
    "磁",
    "貴",
    "互",
    "抵",
    "誉",
    "滅",
    "威",
    "垣",
    "滋",
    "丘",
    "岳",
    "巧",
    "后",
    "瓶",
    "慈",
    "邸",
    "恒",
    "遭",
    "倫",
    "蒙",
    "賓",
    "槽",
    "朽",
    "曹",
    "寡",
  ],
  "radical_name": "一",
}
`;
